exotica_core/src/visualization_meshcat.cpp

```diff
@@ -0,0 +1,342 @@
+                        }
+                        else
+                        {
+                            HIGHLIGHT("Unsupported shape! " << element->segment.getName());
+                        }
+                    }
+                }
+                break;
+            }
+        }
+    }
+}
+
+void VisualizationMeshcat::DisplayState(Eigen::VectorXdRefConst state, double t)
+{
+    const std::vector<std::weak_ptr<KinematicElement>>& elements = scene_->GetKinematicTree().GetTree();
+    scene_->Update(state, t);
+
+    for (std::weak_ptr<KinematicElement> weak_element : elements)
+    {
+        std::shared_ptr<KinematicElement> element = weak_element.lock();
+        if (element->visual.size() == 0) continue;
+        for (auto visual : element->visual)
+        {
+            SendMsg(visualization::SetTransform(path_prefix_ + visual.name, FrameToVector(element->frame)));
+        }
+    }
+}
+
+void VisualizationMeshcat::DisplayTrajectory(Eigen::MatrixXdRefConst trajectory, double dt)
+{
+    auto set_animation = visualization::SetAnimation();
+
+    std::string clip_name = "default";
+    double fps = 1.0 / dt;
+
+    const std::vector<std::weak_ptr<KinematicElement>>& elements = scene_->GetKinematicTree().GetTree();
+    for (std::weak_ptr<KinematicElement> weak_element : elements)
+    {
+        std::shared_ptr<KinematicElement> element = weak_element.lock();
+        if (element->visual.size() == 0) continue;
+        for (auto visual : element->visual)
+        {
+            visualization::Animation anim(path_prefix_ + visual.name);
+            anim.clip = visualization::Clip(fps, clip_name);
+            visualization::Track position(".position", "vector3");
+            visualization::Track rotation(".quaternion", "quaternion");
+            anim.clip.tracks.push_back(position);
+            anim.clip.tracks.push_back(rotation);
+            set_animation.animations.push_back(anim);
+        }
+    }
+
+    for (int t = 0; t < trajectory.rows(); ++t)
+    {
+        double time = (double)t * dt;
+        scene_->Update(trajectory.row(t), time);
+
+        int i = 0;
+        for (std::weak_ptr<KinematicElement> weak_element : elements)
+        {
+            std::shared_ptr<KinematicElement> element = weak_element.lock();
+            if (element->visual.size() == 0) continue;
+            for (auto visual : element->visual)
+            {
+                set_animation.animations[i].clip.tracks[0].keys.push_back(visualization::Key(time, PositionToVector(element->frame)));
+                set_animation.animations[i].clip.tracks[1].keys.push_back(visualization::Key(time, QuaternionToVector(element->frame)));
+                i++;
+            }
+        }
+    }
+
+    SendMsg(set_animation);
+}
+
+void VisualizationMeshcat::Delete(const std::string& path)
+{
+    SendMsg(visualization::Delete("/exotica/" + path));
```

Comment:
Hard-coded prefix

Reply:
We can open it up in the future. It is only preventing deletion of non-exotica objects for now.